encoder.go

```diff
@@ -90,21 +91,26 @@ func encodeStruct(val reflect.Value) ([]byte, error) {
 
 	t := val.Type()
 	for i := 0; i < t.NumField(); i++ {
-		b.WriteString("<member>")
 		f := t.Field(i)
 
 		name := f.Tag.Get("xmlrpc")
+		// if the tag has the omitempty property, skip it
+		if strings.HasSuffix(name, ",omitempty") && val.FieldByName(f.Name).IsZero() {
```

Comment:
You can do `val.Field(i).IsZero()` here. Also, assign the result of `val.Field(i)` to variable and re-use it in the `encodeValue` call.

Reply:
`encodeValue()` later was already using `val.FieldByName()`, so I reused what was there to avoid stepping on toes.

I've saved and reused it. I also went ahead and renamed the other related elements of the struct for clarity.